packages/tsx-editor/src/components/Editor.tsx

```diff
@@ -1,12 +1,19 @@
 import * as monaco from 'monaco-editor/esm/vs/editor/editor.api';
-import React from 'react';
+import * as React from 'react';
 import { IEditorProps } from './Editor.types';
 
-export class Editor extends React.Component<IEditorProps> {
-  private editor: monaco.editor.IStandaloneCodeEditor | undefined;
-  private editorRef = React.createRef<HTMLDivElement>();
+export const Editor = (props: IEditorProps) => {
+  const ref = React.useRef<HTMLDivElement>(null);
+  const { width, height, onChange, code, language } = props;
+  const style = { width, height };
+
+  React.useEffect(() => {
+    const editor = monaco.editor.create(ref.current!, {
```

Comment:
Nit: `ref.current` may well be `null` here; is that an issue? If so you could make it a condition of evaluating the effect and only run this code if it's actually defined.

Reply:
I believe since useEffect is only called on mount, it's going to be in the dom and current will point to the correct element. From my understanding, this shouldn't be an issue, but please correct me if I'm wrong.